src/core/chunked_stream.js

```diff
@@ -267,24 +242,483 @@ var ChunkedStream = (function ChunkedStreamClosure() {
         return missingChunks;
       };
       var subStream = new ChunkedStreamSubstream();
+      subStream.ensureRange(start, start + length);
       subStream.pos = subStream.start = start;
       subStream.end = start + length || this.end;
       subStream.dict = dict;
+      subStream.totalLength = subStream.end - subStream.start;
+      subStream.subStream = true;
+
+      if (subStream.start >= subStream.buffer.start &&
+          subStream.end <= subStream.buffer.end) {
+        // Stream is fully loaded, use fast getByteMethod
+        subStream.getByte = this.continuousAlloc ?
+          this.getByteFastContinuous(subStream.buffer.buffer, subStream.end) :
```

Comment:
create a method with e.g. `createGetByteFast` and implement is differently on the subclasses